htdocs/web_portal/controllers/user/view_user.php

```diff
@@ -20,147 +21,237 @@
+    $params = [];
+
+    validateUserID();
+
+    list($user, $callingUser, $params) = getUserAndUserService();
+    // Check if user only has one identifier to disable unlinking
+    $params['lastIdentifier'] = (count($user->getUserIdentifiers()) === 1);
+    // Add the display locations of the policy files.
+    getPolicyURLs($params);
+
+    $params['user'] = $user;
+    $params['callingUser'] = $callingUser;
+
+    list($roleProjectIds) = getRoleProjectIDs($user, $callingUser);
+
+    $currentIdString = Get_User_Principle();
+    $params['currentIdString'] = $currentIdString;
+
+    /**
+     * Get a list of the projects and their Ids for grouping roles
+     * by project in view.
+     */
+    $projectNamesIds = array();
+    $projects = \Factory::getProjectService()->getProjects();
+
+    foreach ($projects as $project) {
+        $projectNamesIds[$project->getId()] = $project->getName();
+    }
+
+    /**
+     * Check to see if the current calling user has permission to
+     * edit the target user.
+     */
+    $params['projectNamesIds'] = $projectNamesIds;
+    $params['role_ProjIds'] = $roleProjectIds;
+    $params['portalIsReadOnly'] = \Factory::getConfigService()->IsPortalReadOnly();
+    $params['APIAuthEnts'] = $user->getAPIAuthenticationEntities();
+    $title = $user->getFullName();
+
+    show_view("user/view_user.php", $params, $title);
+}
+
+function validateUserID()
+{
     if (!isset($_GET['id']) || !is_numeric($_GET['id'])) {
         throw new Exception("An id must be specified");
     }
+}
 
+/**
+ * Helper function to get user(All User's) information, viewing/calling user,
+ * and few parameters for display.
+ *
+ * @return array An array containing user, viewing/calling user, and
+ *               additional parameters.
+ */
+function getUserAndUserService()
+{
+    $childParams = [];
     $userService = \Factory::getUserService();
     $user = $userService->getUser($_GET['id']);
 
     if ($user === null) {
         throw new Exception("No user with that ID");
     }
 
-    $params = array();
-
-    // Check if user only has one identifier to disable unlinking
-    $params['lastIdentifier'] = (count($user->getUserIdentifiers()) === 1);
-    // Add the display locations of the policy files.
-    getPolicyURLs($params);
+    $callingUser = isUserAuthorizedToView($user, $userService);
 
-    $apiAuthEnts = $user->getAPIAuthenticationEntities();
-    $authEntSites = array();
-    /** @var \APIAuthentication $apiAuth */
-    foreach ($apiAuthEnts as $apiAuth) {
-        $authEntSites[$apiAuth->getParentSite()->getId()]++;
+    try {
+        $userService->editUserAuthorization($user, $callingUser);
+        $childParams['ShowEdit'] = true;
+    } catch (Exception $e) {
+        $childParams['ShowEdit'] = false;
     }
 
+    $childParams['idString'] = $userService->getDefaultIdString($user);
+
+    return [$user, $callingUser, $childParams];
+}
+
+/**
+ * Validates whether the viewing/calling user is authorized to perform
```

Comment:
the word 'authorized' is used here, but in the comment on line 129 it is spelt 'authorised'. Authorised spelt with an S is the British way of spelling it, whilst with a Z is the American way. It should probably be changed here and in the lines below to be consistent with the British way of spelling it. (https://grammarist.com/spelling/authorise-authorize/)

Reply:
Done. Followed the British way.